### PR TITLE
print requirement info

### LINKED ISSUE #6: Tools Requirements
### Please describe your feature request:
A few tools might have requirements (shared libraries/file permissions - ref https://github.com/projectdiscovery/pdtm-api/issues/3)

### Describe the use case of this feature:
The user should be warned of missing dependencies and/or suggested the command to run to fix the issue (eg. `apt/brew install lipbcap-dev`)

### PR DESCRIPTION
**Pull request by dogancanbakir**
This PR prints tool requirement info for given tools. Closes #6.

### CODE REVIEW

File: internal/runner/runner.go [hunk 1]
```diff
@@ -146,6 +152,45 @@ func (r *Runner) Run() error {
...
+func printRequirementInfo(tool *types.Tool) {
+	specs := getSpecs(tool)
+
+	printTitle := true
+	for _, spec := range specs {
+		if requirementSatisfied(spec.Name) {
+			continue
+		}
+		if printTitle {
+			fmt.Printf("%s\n", au.Bold(tool.Name+" requirements:").String())
```
Comment (Mzack9999) [inline internal/runner/runner.go:164]: I think we should stick to `gologger` if it's not causing issues with coloring

File: internal/runner/runner.go [hunk 2]
```diff
@@ -146,6 +152,45 @@ func (r *Runner) Run() error {
...
+			printTitle = false
+		}
+		instruction := getFormattedInstruction(spec)
+		isRequired := getRequirementStatus(spec)
+		fmt.Printf("%s %s\n", isRequired, instruction)
```
Comment (Mzack9999) [inline internal/runner/runner.go:169]: same here

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
suggesting `fmt.xxx` => `gologger`

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
It seems like there is some issue with deps detection on osx:
```
$ ./pdtm -i naabu

                ____          
     ____  ____/ / /_____ ___ 
    / __ \/ __  / __/ __ __  \
   / /_/ / /_/ / /_/ / / / / /
  / .___/\__,_/\__/_/ /_/ /_/ 
 /_/                         

                projectdiscovery.io

[INF] Current pdtm version v0.0.7 (latest)
[INF] Run `source ~/.bashrc` to add /Users/marcornvh/.pdtm/go/bin to $PATH 
[INF] installing naabu...
[INF] installed naabu 2.1.6 (latest)
[INF] naabu requirements:
required To install libpcap on macOS, run the following command in the terminal: sudo brew install libpcap
$ brew install libpcap
Warning: libpcap 1.10.4 is already installed and up-to-date.
```

We can try a more straightforward strategy, like executing the binary just downloaded and checking the output for errors, then suggesting how to install deps. What do you think?

**Comment by dogancanbakir** [comment]
> It seems like there is some issue with deps detection on osx:
> 
> ```
> $ ./pdtm -i naabu
> 
>                 ____          
>      ____  ____/ / /_____ ___ 
>     / __ \/ __  / __/ __ __  \
>    / /_/ / /_/ / /_/ / / / / /
>   / .___/\__,_/\__/_/ /_/ /_/ 
>  /_/                         
> 
>                 projectdiscovery.io
> 
> [INF] Current pdtm version v0.0.7 (latest)
> [INF] Run `source ~/.bashrc` to add /Users/marcornvh/.pdtm/go/bin to $PATH 
> [INF] installing naabu...
> [INF] installed naabu 2.1.6 (latest)
> [INF] naabu requirements:
> required To install libpcap on macOS, run the following command in the terminal: sudo brew install libpcap
> $ brew install libpcap
> Warning: libpcap 1.10.4 is already installed and up-to-date.
> ```
> 
> We can try a more straightforward strategy, like executing the binary just downloaded and checking the output for errors, then suggesting how to install deps. What do you think?

It appears that the issue may be related to the naming and extension convention specific to the operating system. See https://github.com/projectdiscovery/utils/blob/main/syscallutil/syscallutil_test.go#L15 for more information. We can proceed with your suggestion as it seems more convenient.

**Comment by Mzack9999** [comment]
@dogancanbakir I think we can improve the multi-platform check and make it more precise with additional steps (like running the tool and seeing if any error gets printed to stdout/stderr - as previously mentioned. I changed red to yellow for now since the missing dependency might prevent the tool from running under certain conditions but not being installed as we are downloading the binary directly.

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**